src/loaders/lottie/tvgLottieLoader.cpp

```diff
@@ -320,33 +320,81 @@ bool LottieLoader::frame(float no)
     no = roundf(no);
     no *= 0.001f;
 
-    this->frameNo = no;
+    this->frameNo = no + startFrame();
 
     TaskScheduler::request(this);
 
     return true;
 }
 
+float LottieLoader::startFrame()
+{
+    return frameCnt * selectedSegmentBegin;
+}
 
 float LottieLoader::totalFrame()
 {
-    return frameCnt;
+    return (selectedSegmentEnd - selectedSegmentBegin) * frameCnt;
 }
 
 
 float LottieLoader::curFrame()
 {
-    return frameNo;
+    return frameNo - startFrame();
 }
 
 
 float LottieLoader::duration()
 {
-    return frameDuration;
+    if (selectedSegmentBegin == 0.0f && selectedSegmentEnd == 1.0f) return frameDuration;
+
+    if (!comp) done();
+
+    float distance = selectedSegmentEnd - selectedSegmentBegin;
+    float frames = frameCnt * distance;
+    return frames / comp->frameRate;
 }
 
+bool LottieLoader::getSegment(float& start, float& end, const char* marker)
+{
+    if (!marker) {
+        start = selectedSegmentBegin;
```

Comment:
segment(nullptr) will reset the segment.
I think start , end should be 0.0, and 1.0 in this case.

Reply:
When the marker is `nullptr`, this turns out to another usecase of obtaining current segment range set by `LottieAnimation::segment(float begin, float end)`.

So I guess it shouldn't return just `0.0f` and `1.0f` because that's ignoring segment.
Of course, this returns 0.0 and 1.0 when both segment & marker weren't called yet, or reset.

Reply:
Got it.

I just noticed segment(float begin, float end) api has moved in LottieAnimation... It could be in Animation actually. Probably, by this:
![image](https://github.com/thorvg/thorvg/assets/3711518/72c5bd7f-6ed8-4712-91cd-fcac0404635a)

Reply:
@hermet Does the `Result segment(float* start, float* end) noexcept` also have to be in Animation?

Reply:
Please consider this. this is reasonable or not:
```
Animation::segment(float begin, float end);
Animation::segment(float* begin, float* end);
LottieAnimation::segment(const char* marker);
```

If this make sense to you, please update it.

Reply:
This would be clear! Agree